### PR TITLE
Fix warning and usage of ByteArraySet.toArray

### PR DESCRIPTION
**Pull request by qlecorre**
### **GOAL:**

This pull request aims at fixing:
 - a warning (around serialization)
 - an usage of `ByteArraySet.toArray`

### CODE REVIEW

**Comment by eric-weaver** [comment]
Is this change from #75 needed here too?
https://github.com/jwplayer/southpaw/pull/75/files#diff-81d34a13b593abb4f1faa1369448853b3b311903d6d4b2d681c8654fef417928R299

**Comment by qlecorre** [comment]
Good catch, updated

File: src/test/java/com/jwplayer/southpaw/util/ByteArrayTest.java [hunk 1]
```diff
@@ -62,11 +62,12 @@ public void testEquals() {
     @Test
     public void testFromBytes() throws Exception {
         byte[] bytes = Hex.decodeHex("01410204d20100".toCharArray());
-        Set<ByteArray> set = new HashSet<>(3);
-        set.add(new ByteArray("A"));
-        set.add(new ByteArray(1234));
-        set.add(new ByteArray(false));
-        assertEquals(set, ByteArray.fromBytes(bytes));
+        ByteArraySet expectedSet = new ByteArraySet();
```
Comment (eric-weaver) [inline src/test/java/com/jwplayer/southpaw/util/ByteArrayTest.java:65]: hmm @moritastic does it make sense for `ByteArraySet` to be used to test `ByteArray` here?

**Review by eric-weaver — APPROVED** [review]
Changes look good to me! Left one minor question about the `ByteArray` test but that's more thinking from a test boundary standpoint

File: src/test/java/com/jwplayer/southpaw/util/ByteArrayTest.java (hunk 1 above)
Comment (moritastic) [reply src/test/java/com/jwplayer/southpaw/util/ByteArrayTest.java:65]: Yeah, it seems like `toBytes` and `fromBytes` should be on `ByteArraySet`. Not sure why I put it on `ByteArray`.

**Review by moritastic — APPROVED** [review]

File: src/main/java/com/jwplayer/southpaw/util/ByteArray.java [hunk 2]
```diff
@@ -31,6 +31,7 @@
  * Wrapper class for byte arrays so we can use them as keys in maps.
  */
 public class ByteArray implements Comparable<ByteArray>, Serializable {
+    private static final long serialVersionUID = -6277178128299377659L;
```
Comment (arron-green) [inline src/main/java/com/jwplayer/southpaw/util/ByteArray.java:34]: i cant find any references to this variable. If nothing uses it then maybe we should delete it?

File: src/main/java/com/jwplayer/southpaw/util/ByteArraySet.java [hunk 3]
```diff
@@ -585,6 +585,23 @@ public boolean retainAll(Collection<?> c) {
         throw new NotImplementedException();
     }
 
+    @Override
+    public boolean equals(Object object) {
```
Comment (arron-green) [inline src/main/java/com/jwplayer/southpaw/util/ByteArraySet.java:589]: locally i deleted this function and tests still pass.  Is this needed?

File: src/main/java/com/jwplayer/southpaw/util/ByteArray.java (hunk 2 above)
Comment (arron-green) [reply src/main/java/com/jwplayer/southpaw/util/ByteArray.java:34]: nevermind. read the docs here
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Serializable.html

File: src/main/java/com/jwplayer/southpaw/util/ByteArraySet.java (hunk 3 above)
Comment (qlecorre) [reply src/main/java/com/jwplayer/southpaw/util/ByteArraySet.java:589]: I added this function to mimic the way `equals` work for sets in Java. This does not help fixing the bug, may is a nice to have I think.

Comment (eric-weaver) [reply src/main/java/com/jwplayer/southpaw/util/ByteArraySet.java:589]: I have some concerns with adding this. I don't see this being necessary and actually calling [contains](https://github.com/jwplayer/southpaw/blob/master/src/main/java/com/jwplayer/southpaw/util/ByteArraySet.java#L148) for each item is potentially very expensive as it will iterate through the set for each item in the set.

**Review by eric-weaver — APPROVED** [review]

**Review by arron-green — APPROVED** [review]
